lib/si-split-graph/src/lib.rs

```diff
@@ -0,0 +1,1343 @@
+use std::collections::{HashSet, VecDeque};
+
+use async_trait::async_trait;
+use opt_zip::OptZip;
+use petgraph::{prelude::*, stable_graph};
+use serde::{Deserialize, Serialize};
+use si_events::{
+    merkle_tree_hash::MerkleTreeHash,
+    workspace_snapshot::{Change, EntityKind},
+    ContentHash,
+};
+use si_id::ulid::Ulid;
+use thiserror::Error;
+
+mod opt_zip;
+pub mod subgraph;
+pub mod subgraph_address;
+pub mod updates;
+
+use subgraph::{SubGraph, SubGraphEdgeIndex, SubGraphNodeIndex};
+pub use subgraph_address::SubGraphAddress;
+use updates::Update;
+
+pub const MAX_NODES: usize = ((u16::MAX / 2) - 1) as usize;
```

Comment:
How did we get to this value?

Reply:
arbitrary. however i think we're going to move towards providing a closure to the graph that decides when to split